### PR TITLE
The task has done

### PR DESCRIPTION
**Pull request by NekitCool1**

### CODE REVIEW

File: src/main/java/book/shop/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
+import book.shop.validation.Author;
+import book.shop.validation.Unique;
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotNull;
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class CreateBookRequestDto {
+    @NotNull
```
Comment (Elena-Bruyako) [inline src/main/java/book/shop/dto/CreateBookRequestDto.java]:
```suggestion
    @NotBlank
```

check other String

File: src/main/java/book/shop/exception/CustomGlobalExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        body.put("errors", errors);
+        return new ResponseEntity<>(body, headers, status);
+    }
+
+    @ExceptionHandler(EntityNotFoundException.class)
+    public ResponseEntity<Object> handleEntityNotFoundException(EntityNotFoundException e) {
+        String message = String.format("%s, %s", LocalDateTime.now()
+                .format(DateTimeFormatter.ofPattern("yyyy.MM.dd HH:mm")), e.getMessage());
+        Response response = new Response(message);
+        return new ResponseEntity<>(response, HttpStatus.BAD_REQUEST);
```
Comment (Elena-Bruyako) [inline src/main/java/book/shop/exception/CustomGlobalExceptionHandler.java]:
```suggestion
        return new ResponseEntity<>(response, HttpStatus.NOT_FOUND);
```

File: src/main/java/book/shop/dto/Response.java [hunk 3]
```diff
@@ -0,0 +1,4 @@
+package book.shop.dto;
+
+public record Response(String message) {
```
Comment (Elena-Bruyako) [inline src/main/java/book/shop/dto/Response.java]: Too general name, maybe BookApiResponse ?

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/book/shop/dto/CreateBookRequestDto.java [hunk 4]
```diff
@@ -1,14 +1,26 @@
...
 
 @Data
 public class CreateBookRequestDto {
+    @NotBlank
     private String title;
+    @NotBlank
+    @Author
     private String author;
+    @NotNull
+    @Min(0)
```
Comment (MentorMateAcademy) [inline src/main/java/book/shop/dto/CreateBookRequestDto.java]:
```suggestion
    @Positive
```

File: src/main/java/book/shop/dto/CreateBookRequestDto.java [hunk 5]
```diff
@@ -1,14 +1,26 @@
...
     private BigDecimal price;
     private String description;
```
Comment (MentorMateAcademy) [inline src/main/java/book/shop/dto/CreateBookRequestDto.java:23]: you can validate description

File: src/main/java/book/shop/exception/CustomGlobalExceptionHandler.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+            WebRequest request
+    ) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now()
+                .format(DateTimeFormatter.ofPattern("yyyy.MM.dd HH:mm")));
+        body.put("status", HttpStatus.BAD_REQUEST);
+        List<String> errors = ex.getBindingResult().getAllErrors().stream()
+                .map(this::getErrorMessage)
+                .toList();
+        body.put("errors", errors);
```
Comment (MentorMateAcademy) [inline src/main/java/book/shop/exception/CustomGlobalExceptionHandler.java]: Save strings to constants

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]

File: src/main/java/book/shop/dto/CreateBookRequestDto.java (hunk 5 above)
Comment (okuzan) [reply src/main/java/book/shop/dto/CreateBookRequestDto.java:23]: description is not required. Let's validate size instead. Unique is a bit weird here

**Review by okuzan — CHANGES_REQUESTED** [review]

**Pull request merged**